### PR TITLE
mon/MonCommands: "smart" only needs read permission

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
```
2019-10-14T07:31:58.833+0000 7f20c62d7700  1 -- [v2:172.21.15.5:3300/0,v1:172.21.15.5:6789/0] --> 172.21.15.5:0/12358 -- command_reply(tid 33: -1 insufficient caps) v1 -- 0x56189
759f340 con 0x5618979a4800
2019-10-14T07:31:58.833+0000 7f20cdb75700 -1 asok(0x561895dd8000) AdminSocket: request '[{"prefix": "smart", "devid": "INTEL_SSDPED1D480GA_PHMB751300EW480DGN", "format": "json"}]' not defined
2019-10-14T07:31:58.833+0000 7f20cdb75700  1 -- [v2:172.21.15.5:3300/0,v1:172.21.15.5:6789/0] --> 172.21.15.5:0/12358 -- command_reply(tid 33: -22 unknown command prefix smart) v1 -- 0x561897446000 con 0x5618979a4800
```

this change should address the "insufficient caps" issue.

File: src/mon/MonCommands.h [hunk 1]
```diff
@@ -1255,4 +1255,4 @@ COMMAND("config generate-minimal-conf",
 
 COMMAND_WITH_FLAG("smart name=devid,type=CephString,req=false",
 		  "Query health metrics for underlying device",
-		  "mon", "rw", FLAG(HIDDEN))
+		  "mon", "r", FLAG(HIDDEN))
```
Comment (tchaikov) [inline src/mon/MonCommands.h]: i think it conflicts with #30859

**Comment by tchaikov** [comment]
jenkins test submodules

**Comment by liewegas** [comment]
this may be useful for nautilus, but will be obsolete in master as of #30859

**Comment by tchaikov** [comment]
closing to rename the branch name so it contains "nautilus". see #30909

**Pull request closed without merging**